src/structure/structure.ts

```diff
@@ -254,4 +338,239 @@ export class StructureViewer {
         }
         return this._cachedStructures[index];
     }
+
+    /**
+     * Returns the GUID string corresponding to the selected widget.
+     */
+    public get active() {
+        return this._active;
+    }
+
+    /*
+     * Function to set the active widget for communicating with the map
+     */
+    public set active(activeGUID: string) {
+        /// this is here to prevent infinite loops
+        if (activeGUID !== this._active) {
+            let indexes;
+
+            if (this._selected.has(activeGUID) ) {
+                if (this._selected.has(this._active)) {
+                    const oldButton = getByID(`chsp-activate-${this._active}`);
+                    oldButton.classList.toggle('chsp-inactive-structure-marker', true);
+                    oldButton.classList.toggle('chsp-active-structure-marker', false);
+                    oldButton.innerHTML = `<span class="tooltiptext">Choose as active.</span>`;
+                }
+                this._active = activeGUID;
+                const newButton = getByID(`chsp-activate-${this._active}`);
```

Comment:
Same as in the map, the class toggling logic could be extracted to a separate `GridCell` class, with appropriate functions.

This can be refactored later